src/main/java/com/google/sps/data/GivrUser.java

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.QueryResultList;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+
+public final class GivrUser {
+  private String id;
+  private boolean isMaintainer = false;
+
+  public GivrUser(String id, boolean isMaintainer) {
+    this.id = id;
+    this.isMaintainer = isMaintainer;
+  }
+
+  public boolean isMaintainer() {
+    return this.isMaintainer;
+  }
+
+  public static GivrUser getUserByIdFromDatastore(String userId) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Filter queryFilter = new FilterPredicate("userId", FilterOperator.EQUAL, userId);
+    Query query = new Query("User").setFilter(queryFilter);
+      
+    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(1);
+    PreparedQuery preparedQuery = datastore.prepare(query);
+    QueryResultList<Entity> userResult = preparedQuery.asQueryResultList(fetchOptions);
+
+    boolean isMaintainer = false;
+
+    if (userResult.size() == 1) {
+      for (Entity entity: preparedQuery.asIterable(fetchOptions)) {
+        isMaintainer = (boolean) entity.getProperty("isMaintainer");
+      }
+    }
+    GivrUser user = new GivrUser(userId, isMaintainer);
+    return user;
+  }
+
+  public static GivrUser getUserByEmail(String email) {
+    String authDomain = "gmail.com";
```

Comment:
If you intend to **ever** support oauth, I'd suggest opening a tracking issue: "Support OAuth", and add a todo here that links to that issue

That way, you will be easily be able to find the places in the code where you assumed google auth.

Reply:
I'll open an issue, but I'm not sure if we will be able to get it working within the next four weeks :)